### ISSUE TITLE
OutOfMemoryException

### ISSUE THREAD

**Issue by igitur** [Fody/PropertyChanged#125]
Hi,

I have a resources constrained PC, so any memory leak is quickly an issue. I suspect there is some kind of memory leak, but possibly in Cecil.

When I build I regularly get this. Only after restarting Visual Studio 2010 does it work again:
```
    Fody (version 1.28.3.0) Executing
MSBUILD : error : Fody: An unhandled exception occurred:
MSBUILD : error : Exception:
MSBUILD : error : Could not resolve 'System.Windows.DependencyObject'.
MSBUILD : error : StackTrace:
MSBUILD : error :    at ModuleWeaver.InnerResolve(TypeReference reference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\TypeResolver.cs:line 27
MSBUILD : error :    at ModuleWeaver.Resolve(TypeReference reference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\TypeResolver.cs:line 16
MSBUILD : error :    at ModuleWeaver.HierarchyImplementsINotify(TypeReference typeReference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\NotifyInterfaceFinder.cs:line 21
MSBUILD : error :    at ModuleWeaver.HierarchyImplementsINotify(TypeReference typeReference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\NotifyInterfaceFinder.cs:line 44
MSBUILD : error :    at ModuleWeaver.HierarchyImplementsINotify(TypeReference typeReference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\NotifyInterfaceFinder.cs:line 44
MSBUILD : error :    at ModuleWeaver.HierarchyImplementsINotify(TypeReference typeReference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\NotifyInterfaceFinder.cs:line 44
MSBUILD : error :    at ModuleWeaver.HierarchyImplementsINotify(TypeReference typeReference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\NotifyInterfaceFinder.cs:line 44
MSBUILD : error :    at ModuleWeaver.HierarchyImplementsINotify(TypeReference typeReference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\NotifyInterfaceFinder.cs:line 44
MSBUILD : error :    at ModuleWeaver.HierarchyImplementsINotify(TypeReference typeReference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\NotifyInterfaceFinder.cs:line 44
MSBUILD : error :    at ModuleWeaver.HierarchyImplementsINotify(TypeReference typeReference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\NotifyInterfaceFinder.cs:line 44
MSBUILD : error :    at ModuleWeaver.PopulateINotifyNodes(List`1 typeNodes) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\TypeNodeBuilder.cs:line 38
MSBUILD : error :    at ModuleWeaver.BuildTypeNodes() in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\TypeNodeBuilder.cs:line 26
MSBUILD : error :    at ModuleWeaver.Execute() in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\ModuleWeaver.cs:line 27
MSBUILD : error :    at lambda_method(Closure , Object )
MSBUILD : error :    at InnerWeaver.ExecuteWeavers() in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 144
MSBUILD : error :    at InnerWeaver.Execute() in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 62
MSBUILD : error : Source:
MSBUILD : error : PropertyChanged.Fody
MSBUILD : error : TargetSite:
MSBUILD : error : Mono.Cecil.TypeDefinition InnerResolve(Mono.Cecil.TypeReference)
MSBUILD : error : Could not read 'C:\Program Files\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\WindowsBase.dll'.
MSBUILD : error : StackTrace:
MSBUILD : error :    at InnerWeaver.GetAssembly(String file, ReaderParameters parameters) in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\AssemblyResolver.cs:line 30
MSBUILD : error :    at InnerWeaver.Resolve(AssemblyNameReference assemblyNameReference, ReaderParameters parameters) in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\AssemblyResolver.cs:line 49
MSBUILD : error :    at InnerWeaver.Resolve(AssemblyNameReference assemblyNameReference) in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\AssemblyResolver.cs:line 36
MSBUILD : error :    at Mono.Cecil.MetadataResolver.Resolve(TypeReference type)
MSBUILD : error :    at Mono.Cecil.TypeReference.Resolve()
MSBUILD : error :    at ModuleWeaver.InnerResolve(TypeReference reference) in c:\TeamCity\buildAgent\work\eaa0f905cc364117\PropertyChanged.Fody\TypeResolver.cs:line 23
MSBUILD : error : Source:
MSBUILD : error : FodyIsolated
MSBUILD : error : TargetSite:
MSBUILD : error : Mono.Cecil.AssemblyDefinition GetAssembly(System.String, Mono.Cecil.ReaderParameters)
MSBUILD : error : Exception of type 'System.OutOfMemoryException' was thrown.
MSBUILD : error : StackTrace:
MSBUILD : error :    at Mono.Cecil.PE.ImageReader.ReadSectionData(Section section)
MSBUILD : error :    at Mono.Cecil.PE.ImageReader.ReadSections(UInt16 count)
MSBUILD : error :    at Mono.Cecil.PE.ImageReader.ReadImage()
MSBUILD : error :    at Mono.Cecil.PE.ImageReader.ReadImageFrom(Stream stream)
MSBUILD : error :    at Mono.Cecil.ModuleDefinition.ReadModule(Stream stream, ReaderParameters parameters)
MSBUILD : error :    at Mono.Cecil.ModuleDefinition.ReadModule(String fileName, ReaderParameters parameters)
MSBUILD : error :    at InnerWeaver.GetAssembly(String file, ReaderParameters parameters) in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\AssemblyResolver.cs:line 25
MSBUILD : error : Source:
MSBUILD : error : Mono.Cecil
MSBUILD : error : TargetSite:
MSBUILD : error : Void ReadSectionData(Mono.Cecil.PE.Section)
MSBUILD : error : 
```

**Comment by SimonCropp** [comment]
can you upload a solution that this occurs in

**Comment by igitur** [comment]
I'll try to get a sample solution that replicates the error. Unfortunately I can't upload my current solution as it's in-house stuff.

**Comment by SimonCropp** [comment]
feel free to open a new issue when you have a repro

**Comment by gjhdigital** [comment]
Hello, I realize this is marked as closed but Im getting the System.OutOfMemoryException.
I have a VS2015 Xamarin Forms app with a UWP app as well:
Fody v1.29.4
PropertyChanged.Fody v1.51.3

Any help would be greatly appreciated.

Severity	Code	Description	Project	File	Line	Suppression State
Error		Fody: An unhandled exception occurred:
Exception:
Exception of type 'System.OutOfMemoryException' was thrown.
StackTrace:
   at System.IO.MemoryStream.ToArray()
   at Mono.Cecil.EmbeddedResource.GetResourceData()
   at Mono.Cecil.MetadataBuilder.AddResources()
   at Mono.Cecil.MetadataBuilder.BuildModule()
   at Mono.Cecil.ModuleWriter.<BuildMetadata>b__0(MetadataBuilder builder, MetadataReader _)
   at Mono.Cecil.ModuleDefinition.Read[TItem,TRet](TItem item, Func`3 read)
   at Mono.Cecil.ModuleWriter.BuildMetadata(ModuleDefinition module, MetadataBuilder metadata)
   at Mono.Cecil.ModuleWriter.WriteModuleTo(ModuleDefinition module, Stream stream, WriterParameters parameters)
   at Mono.Cecil.ModuleDefinition.Write(Stream stream, WriterParameters parameters)
   at Mono.Cecil.ModuleDefinition.Write(String fileName, WriterParameters parameters)
   at InnerWeaver.WriteModule() in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\ModuleWriter.cs:line 18
   at InnerWeaver.Execute() in c:\ConsoleBuildAgent\work\ed448661dbb30d2e\FodyIsolated\InnerWeaver.cs:line 85
Source:
mscorlib
TargetSite:
Byte[] ToArray()	RTRGF.UWP

**Comment by jasin755** [comment]
Please reopen this issue. I have same problem. This is simple project: https://gitlab.com/jasin755/fodybug

**Comment by kolchy** [comment]
Getting the same error in Xam Studio...

**Comment by SimonCropp** [comment]
Can u try the latest version of fody?
